TexturePacker Animation Parsing

Eendhoorn opened the pull request:
- Added support for importing TexturePacker regions' pivot points. Especially useful when using "Trim Mode - Crop, keep position" in TexturePacker
- Implemented automagic animation parsing based on TexturePacker region filenames

Not sure what's going on with SubTexture.cs I tried merging a million times but it keeps saying there are changes :/

#109

prime31 commented:
This all looks good with one minor exception: it looks like the Subtexture `center` field was reused as a texture origin which makes the name misleading.

I'll have to give some thought on how to best handle the origin. Currently SpriteAnimationFrame is where the origin lives. I am leaning towards removing SpriteAnimationFrame altogether and just adding an `origin` field to Subtexture. That would reduce the object count which is always good and also solve this particular case.

If you have any suggestions on how you would solve this feel free to post them here.

prime31 commented:
OK. So, I pushed commit 0d486b4 which should provide all the tooling to get origin working. Subtexture now has an origin field that is used throughout the Sprite classes. SpriteAnimationFrame was killed since Subtexture now has origin.

Eendhoorn commented:
Ahh I did mistake the center and origin here, sorry. I have pushed some commits where I removed the SubTexture constructors which were setting the center instead of the origin, and adjusted TexturePackerAtlas to reflect the change.

Seems to working good now!

One thing I should note is right now is that TexturePacker needs to have "Sort by : name", to get the frames in the correct order right now. I wasn't quite sure where to document this.
I hope to expand the TexturePackerProcessor in the future so it can handle duplicate frames and all that jazz once I have a bit more time to dive into it.

Thanks!
